integration-test/1810-alternate-viewpoints.py

```diff
@@ -394,3 +393,147 @@ def test_none_none(self):
             'Populated place',
             {}
         )
+
+
+class RegionBoundary(FixtureTest):
+
+    def test_dispute(self):
+        import dsl
+
+        z, x, y = 16, 0, 0
+
+        self.generate_fixtures(
+            # way, tagged disputed
+            dsl.way(1, dsl.tile_diagonal(z, x, y), {
+                'admin_level': '4',
+                'boundary': 'administrative',
+                'disputed': 'yes',
+                'disputed_by': 'XB',
+                'name': 'XA internal region border',
+                'source': 'openstreetmap.org',
+            }),
+            # way used in relations for region borders
+            #
+            # NOTE: we won't use this one, since we don't pay attention to
+            # admin_level=3, but i included it for completeness, so that the
+            # test more closely resembles the data it's based on.
+            dsl.way(2, dsl.tile_diagonal(z, x, y), {
+                'admin_level': '3',  # 3?!? as always... data = mess
+                'border_type': 'province',
+                'boundary': 'administrative',
+                'is_in:country_code': 'XA',
+                'name': 'XA region 1',
+                'type': 'boundary',
+                'mz_boundary_from_polygon': True,  # need this for hack
+            }),
+            # this one we do pay attention to, admin_level=4
+            dsl.way(3, dsl.tile_diagonal(z, x, y), {
+                'admin_level': '4',
+                'boundary': 'administrative',
+                'name': 'XA region 2',
+                'place': 'District',
+                'type': 'boundary',
+                'mz_boundary_from_polygon': True,  # need this for hack
+            }),
+        )
+
+        self.assert_has_feature(
+            z, x, y, 'boundaries', {
+                'kind': 'region',
+                'kind:xb': 'unrecognized_region',
+                'name': 'XA region 2',
```

Comment:
Why isn't this name: `XA internal region border` from the first way? 

Do these need to be linked up in relation or are the loose ways fine? Seems like we sometimes enforce the ways + relations in tests, but not here?

Reply:
We take the name from the relation (boundary or claim), rather than the way (dispute).

The `dsl.way` call is misleading; the generative fixtures used in the tests are injected as if they're the results of a query, and the boundary relation processing is done in the query, so although boundaries are relations (and polygons), the code post-query only sees line strings as if they came from ways.

Apologies - the whole query / RAWR tile input stack has become a mess as we've squeezed more features into it, so now things are not always what they appear (or what the function name suggests).